test/unit/deferred.js

```diff
@@ -525,6 +525,34 @@ QUnit.test( "jQuery.Deferred.then - spec compatibility", function( assert ) {
 	} catch ( _ ) {}
 } );
 
+QUnit[ window.console ? "test" : "skip" ]( "jQuery.Deferred.exceptionHook", function( assert ) {
+
+	assert.expect( 1 );
+
+	var done = assert.async(),
+		defer = jQuery.Deferred(),
+		oldWarn = window.console.warn;
+
+	window.console.warn = function( msg ) {
```

Comment:
The problem with Dexter was the ajax mock. Maybe re-releasing it with only the spy and stub API will serve most of what many libraries want. It does not need to rely on any test framework.

I'll also check if I should even split the spy and stub features in different modules.

Anyway, you can solve this current PR without any mock/spy function as I mentioned in https://github.com/jquery/jquery/pull/2737/files#r46825810